### PR TITLE
Bugfix: Check if expression in CompactInitialization rule should be p…

### LINKED ISSUE #1150: KotlinParseException: Text is not valid: [res as TranslationsSuccess.apply {}]
## Describe the bug
During diktatFix the follow error is logged and the process exits unsuccesfully:
```
[ERROR] 2021-12-15 14:30:23 Internal error has occurred in rule [class-compact-initialization]. Please make an issue on this bug at https://github.com/cqfn/diKTat/. 
                       As a workaround you can disable these inspections in yml config: <[COMPACT_OBJECT_INITIALIZATION]>.
                       Root cause of the problem is in [TranslationAdapterTest.kt] file.
org.cqfn.diktat.ruleset.utils.KotlinParseException: Text is not valid: [res as TranslationsSuccess.apply {}]
```
## Expected behavior
diktatFix successfully formats the file
## Steps to Reproduce
Format Code with sealed class and use `as` for typecasting:
```
package de.core.adapter.driven.translation

import io.kotest.matchers.collections.shouldHaveSize
import io.kotest.matchers.shouldBe
import io.micronaut.http.HttpStatus
import org.junit.jupiter.api.Test

internal class TranslationTest {

    @Test
    fun `translate text`() {
        val res = translateText(text = "dummy")
        (res is TranslationsSuccess) shouldBe true
        val translationsSuccess = res as TranslationsSuccess
        translationsSuccess.translations shouldHaveSize 1
    }

    fun translateText(
        text: String
    ): TranslationsResult = TranslationsSuccess(translations = emptyList())

    internal sealed class TranslationsResult

    internal data class TranslationsSuccess(
        val translations: List<String>
    ) : TranslationsResult()

    internal data class TranslationsFailure(
        val statusCode: HttpStatus,
        val message: String?
    ) : TranslationsResult()
}

```
## Environment information
* diktat version: 1.0.0
* build tool (maven/gradle):gradle
* how is diktat run (CLI, plugin, etc.):    id "org.cqfn.diktat.diktat-gradle-plugin" version "1.0.0"
* kotlin version:1.5.21
* operating system: macOS 12.0.1

### PR DESCRIPTION
**Pull request by petertrr**
…arenthesized

### What's done:
* Fix logic
* Add tests

This pull request closes #1150

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt [hunk 1]
```diff
@@ -145,7 +147,24 @@ class CompactInitialization(configRules: List<RulesConfig>) : DiktatRule(
         ?: run {
             // add apply block
             property.node.run {
-                val newInitializerNode = kotlinParser.createNode("${property.initializer!!.text}.apply {}")
+                val isNotRequiresParentheses = property.initializer.let {
```
Comment (kgevorkyan) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt:150]: May be move this negation into the code below? Thus, we will have 
```kotlin
val isRequireParentheses = property.initializer.let {
!(it is KtParenthesizedExpression || it is KtDotQualifiedExpression ||
                            it is KtCallExpression || it is KtReferenceExpression)
}
````

And further 
```kotlin
if (isRequireParentheses) {
   append("(")
}
```

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt [hunk 2]
```diff
@@ -145,7 +147,24 @@ class CompactInitialization(configRules: List<RulesConfig>) : DiktatRule(
...
+                    // list of expression types, that can be directly followed by a dot-qualified call
+                    // e.g. val x = foo()  ->  val x = foo().apply {}
+                    // e.g. val x = foo + bar  ->  val x = (foo + bar).apply {}
+                    it is KtParenthesizedExpression || it is KtDotQualifiedExpression ||
+                            it is KtCallExpression || it is KtReferenceExpression
```
Comment (kgevorkyan) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt:155]: What about safe access expressions?

May be we can add more tests?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt [hunk 3]
```diff
@@ -145,7 +147,24 @@ class CompactInitialization(configRules: List<RulesConfig>) : DiktatRule(
...
+                }
+                val newInitializerNodeText = buildString {
+                    if (!isNotRequiresParentheses) {
+                        append("(")
+                    }
+                    append(property.initializer!!.text)
+                    if (!isNotRequiresParentheses) {
+                        append(")")
+                    }
+                    append(".apply {}")
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt:165]: I'm sorry, but I didn't get why we are adding empty apply block here...

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt:165]: We are filling it with statements later. This creation is to have a proper AST representation of `apply` block

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt (hunk 1 above)
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt:150]: May be we will move this logic with adding apply block to some new method...
If it looks ok

**Review by akuleshov7 — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt (hunk 1 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt]: Thanks, changed

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/CompactInitialization.kt]: Extracted

**Review by kgevorkyan — APPROVED** [review]

**Pull request merged**